contracts/adapters/AaveFLIStrategyExtension.sol

```diff
@@ -0,0 +1,1277 @@
+        uint256 chunkRebalanceNotional;
+        uint256 totalRebalanceNotional;
+        if (_newLeverageRatio < _leverageInfo.currentLeverageRatio) {
+            (
+                chunkRebalanceNotional,
+                totalRebalanceNotional
+            ) = _calculateChunkRebalanceNotional(_leverageInfo, _newLeverageRatio, false);
+
+            _delever(_leverageInfo, chunkRebalanceNotional); 
+        } else {
+            (
+                chunkRebalanceNotional,
+                totalRebalanceNotional
+            ) = _calculateChunkRebalanceNotional(_leverageInfo, _newLeverageRatio, true);
+
+            _lever(_leverageInfo, chunkRebalanceNotional);
+        }
+
+        return (chunkRebalanceNotional, totalRebalanceNotional);
+    }
+
+    /**
+     * Create the leverage info struct to be used in internal functions
+     *
+     * return LeverageInfo                Struct containing ActionInfo and other data
+     */
+    function _getAndValidateLeveragedInfo(uint256 _slippageTolerance, uint256 _maxTradeSize, string memory _exchangeName) internal view returns(LeverageInfo memory) {
+        // Assume if maxTradeSize is 0, then the exchange is not enabled. This is enforced by addEnabledExchange and updateEnabledExchange
+        require(_maxTradeSize > 0, "Must be valid exchange");
+
+        ActionInfo memory actionInfo = _createActionInfo();
+
+        require(actionInfo.setTotalSupply > 0, "SetToken must have > 0 supply");
+        require(actionInfo.collateralBalance > 0, "Collateral balance must be > 0");
+        require(actionInfo.borrowBalance > 0, "Borrow balance must exist");
+
+        // Get current leverage ratio
+        uint256 currentLeverageRatio = _calculateCurrentLeverageRatio(
+            actionInfo.collateralValue,
+            actionInfo.borrowValue
+        );
+
+        return LeverageInfo({
+            action: actionInfo,
+            currentLeverageRatio: currentLeverageRatio,
+            slippageTolerance: _slippageTolerance,
+            twapMaxTradeSize: _maxTradeSize,
+            exchangeName: _exchangeName
+        });
+    }
+
+    /**
+     * Create the action info struct to be used in internal functions
+     *
+     * return ActionInfo                Struct containing data used by internal lever and delever functions
+     */
+    function _createActionInfo() internal view returns(ActionInfo memory) {
+        ActionInfo memory rebalanceInfo;
+
+        // Calculate prices from chainlink. Chainlink returns prices with 8 decimal places, but we need 36 - underlyingDecimals decimal places 
+        // This is so that when the underlying amount is multiplied by the received price, the collateral valuation is normalized to 36 decimals.
+        // To perform this adjustment, we multiply by 10^(36 - 8 - underlyingDeciamls)
+        int256 rawCollateralPrice = strategy.collateralPriceOracle.latestAnswer();
+        rebalanceInfo.collateralPrice = rawCollateralPrice.toUint256().mul(10 ** 10);
```

Comment:
You have included it above, but you can also mention explicitly here that since chainlink returns prices with only 8 decimal places we multiply by 10 ** 10, to make it 18 decimal places.

Reply:
+1